state-sync/state-sync-v2/state-sync-driver/src/storage_synchronizer.rs

```diff
@@ -385,6 +390,12 @@ fn spawn_executor<ChunkExecutor: ChunkExecutorTrait + 'static>(
                             decrement_pending_data_chunks(pending_transaction_chunks.clone());
                         }
                     }
+
+                    // If the executor begins running too far ahead of the committer
+                    // let's force more yields to avoid unnecessary back pressure.
+                    if load_pending_data_chunks(pending_transaction_chunks.clone()) > max_pending_data_chunks / 2 {
+                        yield_now().await;
+                    }
```

Comment:
Hmm, talking about contention, shouldn't we make `chunk_executor.apply_chunk` and friends who invokes IO operations asynchronous via `spawn_blocking`?

Reply:
Yes, ideally. This is something I called out also in https://github.com/aptos-labs/aptos-core/issues/538.

Note: the executor thread here is technically asynchronous because of its use of select. But, if the queue has pending data it'll keep executing until it hits the max limit 😄